File: imrbs-core/src/main/java/tw/huangcti/imrbs/domain/Reservation.java
package tw.huangcti.imrbs.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;

/**
 * 預約實體
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Reservation {
    private String id;
    private String roomId;
    private LocalDate date;
    private LocalTime startTime;
    private LocalTime endTime;
    private String title;
    private String organizerEmail;
    private List<String> participants; // email 列表
    private ReservationStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String schemaVersion;

    public enum ReservationStatus {
        CONFIRMED,
        CANCELLED
    }
}

File: imrbs-frontend/src/types/reservation.ts
/**
 * 預約相關型別定義
 */

export type ReservationStatus = 'PENDING' | 'CONFIRMED' | 'CANCELLED' | 'COMPLETED'

export interface Reservation {
  id: number
  roomId: number
  roomName?: string
  userId: string
  userName?: string
  startTime: string
  endTime: string
  purpose: string
  participants: string[]
  status: ReservationStatus
  createdAt: string
  updatedAt: string
}

export interface CreateReservationRequest {
  roomId: number
  userId: number
  meetingTitle: string
  startTime: string
  endTime: string
  participants: string
  recurringRule?: string
}

export interface CreateReservationResponse {
  id: number
  message: string
}

export interface UpdateReservationRequest {
  meetingTitle: string
  startTime: string
  endTime: string
  participants: string
}

export interface CancelReservationRequest {
  cancellationReason: string
}
